File: src/main/java/top/zbeboy/isy/web/jcaptcha/CaptchaUtils.java
package top.zbeboy.isy.web.jcaptcha;

import com.octo.captcha.service.CaptchaServiceException;
import lombok.extern.slf4j.Slf4j;

import javax.servlet.http.HttpServletRequest;

/**
 * Created by lenovo on 2016-09-04.
 */
@Slf4j
public class CaptchaUtils {

    /**
     * 验证验证码
     *
     * @param captcha 验证码
     * @param request 请求
     * @return true or false
     */
    public static boolean validCaptcha(String captcha, HttpServletRequest request) {
        Boolean isResponseCorrect = Boolean.FALSE;
        // remenber that we need an id to validate!
        String captchaId = request.getSession().getId();
        // call the service method
        try {
            isResponseCorrect = CaptchaServiceSingleton.getInstance().validateResponseForID(captchaId, captcha);
            return isResponseCorrect;
        } catch (CaptchaServiceException e) {
            log.error("Captcha error : {}", e);
            return Boolean.FALSE;
        }
    }
}

File: src/main/java/top/zbeboy/isy/web/jcaptcha/CaptchaUtils.kt
package top.zbeboy.isy.web.jcaptcha

import com.octo.captcha.service.CaptchaServiceException
import org.slf4j.LoggerFactory
import javax.servlet.http.HttpServletRequest

/**
 * Created by zbeboy 2017-11-20 .
 **/
open class CaptchaUtils {
    companion object {

        private val log = LoggerFactory.getLogger(CaptchaUtils::class.java)

        /**
         * 验证验证码
         *
         * @param captcha 验证码
         * @param request 请求
         * @return true or false
         */
        @JvmStatic
        fun validCaptcha(captcha: String, request: HttpServletRequest): Boolean {
            val isResponseCorrect: Boolean?
            // remenber that we need an id to validate!
            val captchaId = request.session.id
            // call the service method
            try {
                isResponseCorrect = CaptchaServiceSingleton.getInstance().validateResponseForID(captchaId, captcha)
                return isResponseCorrect!!
            } catch (e: CaptchaServiceException) {
                log.error("Captcha error : {}", e)
                return java.lang.Boolean.FALSE
            }

        }
    }
}